lib/services/local/access_list.go

```diff
@@ -57,44 +60,109 @@ func NewAccessListService(backend backend.Backend, clock clockwork.Clock) (*Acce
+	return a.alSvc.ListResources(ctx, pageSize, nextToken)
 }
 
 // GetAccessList returns the specified access list resource.
 func (a *AccessListService) GetAccessList(ctx context.Context, name string) (*accesslist.AccessList, error) {
-	accessList, err := a.svc.GetResource(ctx, name)
+	accessList, err := a.alSvc.GetResource(ctx, name)
 	return accessList, trace.Wrap(err)
 }
 
 // UpsertAccessList creates or updates an access list resource.
 func (a *AccessListService) UpsertAccessList(ctx context.Context, accessList *accesslist.AccessList) (*accesslist.AccessList, error) {
-	if err := trace.Wrap(a.svc.UpsertResource(ctx, accessList)); err != nil {
+	if err := trace.Wrap(a.alSvc.UpsertResource(ctx, accessList)); err != nil {
 		return nil, trace.Wrap(err)
 	}
 	return accessList, nil
 }
 
 // DeleteAccessList removes the specified access list resource.
 func (a *AccessListService) DeleteAccessList(ctx context.Context, name string) error {
-	return trace.Wrap(a.svc.DeleteResource(ctx, name))
+	err := a.alSvc.DeleteResource(ctx, name)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Delete all associated members as well.
+	return trace.Wrap(a.DeleteAllAccessListMembersForAccessList(ctx, name))
 }
 
 // DeleteAllAccessLists removes all access lists.
 func (a *AccessListService) DeleteAllAccessLists(ctx context.Context) error {
-	return trace.Wrap(a.svc.DeleteAllResources(ctx))
+	err := trace.Wrap(a.alSvc.DeleteAllResources(ctx))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Delete all members for all access lists.
+	return trace.Wrap(a.DeleteAllAccessListMembers(ctx))
+}
+
+// GetAccessListMembers returns the members associated with the access list resource.
+func (a *AccessListService) GetAccessListMembers(ctx context.Context, accessList string) ([]*accesslist.AccessListMember, error) {
+	members, err := a.almSvc.GetResources(ctx, accessList)
+	return members, trace.Wrap(err)
+}
+
+// ListAccessListMembers returns a paginated list of all access list members.
+func (a *AccessListService) ListAccessListMembers(ctx context.Context, accessList string, pageSize int, nextToken string) ([]*accesslist.AccessListMember, string, error) {
+	return a.almSvc.ListResources(ctx, pageSize, nextToken, accessList)
+}
+
+// GetAccessListMember returns the specified access list member resource.
+func (a *AccessListService) GetAccessListMember(ctx context.Context, accessList string, memberName string) (*accesslist.AccessListMember, error) {
+	member, err := a.almSvc.GetResource(ctx, memberName, accessList)
+	return member, trace.Wrap(err)
+}
+
+// UpsertAccessListMember creates or updates an access list member resource.
+func (a *AccessListService) UpsertAccessListMember(ctx context.Context, member *accesslist.AccessListMember) (*accesslist.AccessListMember, error) {
+	if err := trace.Wrap(a.almSvc.UpsertResource(ctx, member, member.Spec.AccessList)); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return member, nil
```

Comment:
As a general design comment, it looks off to me that service.UpsertResource doesn't return the member, but this one does. I'd rather the service did it too.

Reply:
I'm amenable to that, but I'd rather that be relegated to its own PR as it would cause a lot of changes for unrelated services in this one.

Reply:
Yep, definitely not something to be changed here. Up to you if you think a follow up is worth it.